src/data/equiv/mul_add.lean

```diff
@@ -45,6 +45,14 @@ def mul_hom.inverse [has_mul M] [has_mul N] (f : mul_hom M N) (g : N → M)
                ... = g (f (g x * g y)) : by rw f.map_mul
                ... = g x * g y : h₁ _, }
 
+/-- The inverse of a bijective `monoid_hom` is a `monoid_hom`. -/
+@[to_additive "The inverse of a bijective `add_monoid_hom` is an `add_monoid_hom`."]
+def monoid_hom.inverse {A B : Type*} [monoid A] [monoid B] (f : A →* B) (g : B → A)
+  (h₁ : function.left_inverse g f) (h₂ : function.right_inverse g f) :
+  B →* A :=
+{ map_one' := show g 1 = 1, by rw [← f.map_one, h₁],
```

Comment:
```suggestion
{ to_fun := g,
  map_one' := by rw [← f.map_one, h₁],
```
First line needed to make simp happy, but may also allow removing the show

Reply:
I believe I understand: the `inverse_apply` lemma generated by `simps` will include a `.to_fun` without this?

IIUC this is not so serious because of `to_fun_eq_coe` but still an improvement, and anyway, it allows dropping the `show` as you mention.